changelogs/fragments/3980-puppet-show_diff.yml

```
@@ -0,0 +1,2 @@
+minor_changes:
+  - puppet - remove deprecation for ``show_diff`` parameter (https://github.com/ansible-collections/community.general/pull/3980).
```

I think we should mention that the alias is still deprecated:
```suggestion
  - puppet - remove deprecation for ``show_diff`` parameter. Its alias ``show-diff`` is still deprecated and will be removed in community.general 7.0.0 (https://github.com/ansible-collections/community.general/pull/3980).
```
(This doesn't need a new `deprecated_features` entry IMO since it was already deprecated before, as part of the whole deprecated option.)

---

Sure, seems clearer.